.changeset/green-mangos-fix.md

```diff
@@ -0,0 +1,12 @@
+---
+'astro': major
+'@astrojs/sitemap': patch
+---
+
+Remove the `pages` parameter from the `astro:build:done` integration hook.
+
+#### Migration
+
+The `pages` parameter was an undocumented parameter used by the `@astrojs/sitemap` integration. If your integration relied on `pages`, use the `routes` parameter and associated `pathname` property instead.
+
+Note: The `pathname` property on `routes` does _not_ respect [the `trailingSlash` property](https://docs.astro.build/en/reference/configuration-reference/#trailingslash) in your Astro config. If needed, ensure you append this slash after reading the user's Astro config.
```

Comment:
Given this is such a small thing is it really worth removing? I understand that there is some overlap but this seems like a difference that some integrations might be relying on.

Reply:
I understand, and maybe we can revert! Though I think having both `pages` and `routes` side-by-side is pretty confusing since they both mean the same thing conceptually. We avoided documenting `pages` in favor of `routes` for this reason. If we keep it around, I'd prefer a clearer name so we can document when it should be used _instead_ of routes.

Reply:
I'd feel better about this if we knew that no one was using it. It's too bad that we don't have telemetry data on this yet (something to add).

I do agree that `pages` is kind of weird, especially that it only applies to SSG. We need to discuss it might though I think.